### ISSUE TITLE
Autowire does not inject monolog

### ISSUE THREAD

**Issue by chinmayshah24** [symfony/symfony#16639]
I am able to inject monolog using this:

```
services:
    service_demo:
        class:  AppBundle\Services\ServiceDemo
    service_consumer_demo:
        class:  AppBundle\Services\ServiceConsumerDemo
        arguments:  ["@doctrine.orm.entity_manager", "@service_demo"]
    controllers.home:
        class:  AppBundle\Controller\HomeController
        arguments:  ["@doctrine.orm.entity_manager", "@service_consumer_demo", "@service_demo", "@monolog.logger"]
```

But not using this:

```
services:
    controllers.home:
        class:  AppBundle\Controller\HomeController
        autowire:   true
```

**Comment by weaverryan** [comment]
@chinmayzen try upgrading your MonologBundle to the laster `dev-master`. This pull request - https://github.com/symfony/monolog-bundle/pull/143 - (which is not part of a release yet), should fix the issue.

**Comment by chinmayzen** [comment]
I have the code in the commit in my MonlogExtension.php. It doesn't work i guess.

```
Warning: Missing argument 1 for Monolog\Logger::__construct()
```

**Comment by stof** [comment]
@chinmayzen what is te typehint in your class ?

**Comment by chinmayzen** [comment]
Symfony\Bridge\Monolog\Logger

**Comment by aitboudad** [comment]
@chinmayzen there are many service that use class `Symfony\Bridge\Monolog\Logger`  use the typehint 'Psr\Log\LoggerInterface' instead to make it work

**Comment by weaverryan** [comment]
I would have expected autowiring to see that `Symfony\Bridge\Monolog\Logger` has many instances and throw an error: not try to instantiate a new one. @dunglas

**Comment by aitboudad** [comment]
fixed #16642

**Comment by chinmayzen** [comment]
Do you guys think it would be beneficial to have something like `@Qualifier` in Spring? That would allow to explicitly specify which service should be injected.. i'm new to Symfony so could obviously be off the mark here
